src/server/saved_objects/migrations/core/build_active_mappings.ts

```diff
@@ -38,14 +39,90 @@ export function buildActiveMappings({
   properties: MappingProperties;
 }): IndexMapping {
   const mapping = defaultMapping();
+
+  properties = validateAndMerge(mapping.doc.properties, properties);
+
   return _.cloneDeep({
     doc: {
       ...mapping.doc,
-      properties: validateAndMerge(mapping.doc.properties, properties),
+      properties,
+      _meta: {
+        migrationMappingPropertyHashes: md5Values(properties),
+      },
     },
   });
 }
 
+/**
+ * Diffs the actual vs expected mappings. The properties are compared using md5 hashes stored in _meta, because
+ * actual and expected mappings *can* differ, but if the md5 hashes stored in actual.doc._meta.migrationMappingPropertyHashes
+ * match our expectations, we don't require a migration. This allows ES to tack on additional mappings that Kibana
+ * doesn't know about or expect, without triggering continual migrations.
+ */
+export function diffMappings(actual: IndexMapping, expected: IndexMapping) {
+  if (actual.doc.dynamic !== expected.doc.dynamic) {
+    return { changedProp: 'doc.dynamic' };
+  }
+
+  if (!actual.doc._meta || !actual.doc._meta.migrationMappingPropertyHashes) {
+    return { changedProp: 'doc._meta' };
+  }
+
+  const changedProp = findChangedProp(
+    actual.doc._meta.migrationMappingPropertyHashes,
+    expected.doc._meta!.migrationMappingPropertyHashes
+  );
+
+  return changedProp ? { changedProp: `doc.properties.${changedProp}` } : undefined;
+}
+
+// Convert an object to an md5 hash string, using rison so we get a stable serialization
```

Comment:
You might want to put a big giant comment here warning that we should never change the output of this implementation or else we will trigger migrations on every single Kibana installation. We might also want tests to confirm this never changes (with some comments that say if this test is failing something is bad).

Reply:
Hm. Yeah. Also, this comment is not right: "using rison" is not longer true.